kf_lib_data_ingest/test_runner.py

```diff
@@ -0,0 +1,130 @@
+"""
+The test runner is responsible for executing unit tests on ingest packages
+
+Test runner always runs the standard tests for an ingest package and
+runs user defined unit tests for a package if they exist.
+"""
+
+import os
+import logging
+import pytest
+import requests_mock
+
+from kf_lib_data_ingest.etl.ingest_pipeline import DataIngestPipeline
+from kf_lib_data_ingest.config import (
+    TARGET_SERVICE_CONFIG_PATH,
+    DEFAULT_RESTRICTED_TARGET_URLS
+)
+
+logger = logging.getLogger(__name__)
+
+
+def run(dataset_ingest_config_path, **kwargs):
+    """
+    Run the standard and user defined unit tests for the ingest package
+    located at `dataset_ingest_config_path`
+
+    :param dataset_ingest_config_path: path to ingest package's config file
+    or the ingest package dir where config file exists
+    :type: str
+    :param kwargs: keyword args that will be forwarded directly to
+    DataIngestPipeline.run
+    :type kwargs: dict
+    """
+    ingest_pipeline = DataIngestPipeline(dataset_ingest_config_path)
+
+    _run_standard_tests(ingest_pipeline, **kwargs)
+    _run_user_defined_tests(ingest_pipeline)
+
+
+def _run_standard_tests(ingest_pipeline, **kwargs):
+    """
+    Runs standard unit tests for the ingest package located at
+    `dataset_ingest_config_path`
+
+    Standard tests:
+        1. Run ingest pipeline in dry run mode. In dry run mode, the ingest
+           pipeline runs without modifying data in the target service
+
+    :param ingest_pipeline: the ingest pipeline containing a reference to the
+    :type ingest_pipeline: DataIngestPipeline
+    :param kwargs: keyword args that will be forwarded directly to
+    DataIngestPipeline.run
+    :type kwargs: dict
+    """
+    logger.info('Running standard unit tests for '
+                f'{ingest_pipeline.ingest_config_dir} ...')
+    # TODO Enable this when dry run mode is implemented
+    # kwargs.update({
+    #     'dry_run': True
+    # })
+    ingest_pipeline.run(TARGET_SERVICE_CONFIG_PATH, **kwargs)
+    logger.info('✅  Standard unit tests passed!')
```

Comment:
What does passing mean here?

Reply:
It's not very clear to me either. Maybe passing means the test runner didn't throw out en error?